core/trino-main/src/main/java/io/trino/operator/PagesIndex.java

```diff
@@ -616,4 +618,12 @@ public Page computeNext()
             }
         };
     }
+
+    public long getEstimatedLookupSourceSizeInBytes(HashArraySizeSupplier hashArraySizeSupplier)
```

Comment:
Estimate will depend on which `PagesHash` implementation is used and is also subject to change in the future.

You could pass `LocalMemoryContext` to `io.trino.operator.PagesIndex#createLookupSourceSupplier` and make `createLookupSourceSupplier` return a future.

Reply:
Discussed offline. Decided to extract helper methods for estimating every part of the JoinHash